### PR TITLE
Three in one for CI

### PR DESCRIPTION
**Pull request by Dekken**
windows pip updates
dropping 3.7 on CI until more mature - i.e. tensorflow release for 3.7 exists
Numpy Pending Deprecation warning breaking tests

### CODE REVIEW

File: tick/array_test/tests/array_memory_test.py [hunk 1]
```diff
@@ -259,15 +259,15 @@ def get_memory_used():
...
-                                       delta=1.1 * bytes_size)
+                                       delta=2.1 * bytes_size)
 
                 X = tick_double_sparse2d_from_file(cereal_file)
                 del X
 
             gc.collect()
 
             end = get_memory_used()
-            self.assertAlmostEqual(end, initial_memory, delta=1.1 * bytes_size)
```
Comment (Mbompr) [inline tick/array_test/tests/array_memory_test.py]: Why this high :( ?

Comment (Dekken) [reply tick/array_test/tests/array_memory_test.py]: tests fail

Comment (Dekken) [reply tick/array_test/tests/array_memory_test.py]: we can make a ticket to investigate it

Comment (Dekken) [reply tick/array_test/tests/array_memory_test.py]: https://github.com/X-DataInitiative/tick/issues/332

Comment (Mbompr) [reply tick/array_test/tests/array_memory_test.py]: That sounds frightening... 
It might mean that each time we create an array we have a memory leak...

Comment (Dekken) [reply tick/array_test/tests/array_memory_test.py]: we haven't changed anything, and it doesn't always happen so I dunno what to tell you

Comment (Dekken) [reply tick/array_test/tests/array_memory_test.py]: I was running python 3.5 so let me test some more

Comment (Dekken) [reply tick/array_test/tests/array_memory_test.py]: works on CI now resetting back to 1.1 delta

**Pull request merged**